src/main/kotlin/app/revanced/patches/youtube/layout/sponsorblock/bytecode/fingerprints/ShortsPlayerConstructorFingerprint.kt

```diff
@@ -33,6 +36,17 @@ object ShortsPlayerConstructorFingerprint : MethodFingerprint(
         Opcode.IPUT_OBJECT,
         Opcode.NEW_INSTANCE,
         Opcode.INVOKE_DIRECT,
-        Opcode.IPUT_OBJECT
-    )
+        Opcode.IPUT_OBJECT,
+        Opcode.IPUT_OBJECT,
+        Opcode.CONST_4
+    ),
+    customFingerprint = { methodDef ->
+        val targetResourceId = ResourceIdMappingProviderResourcePatch.resourceMappings.single {
+            it.type == "id" && it.name == "reel_persistent_edu_button_group"
+        }.id
```

Comment:
This will still call `.single` every time the customFingerprint is being invoked. Move this outside of customFingerprint.

Reply:
> This will still call `.single` every time the customFingerprint is being invoked. Move this outside of customFingerprint.

Please...When you make a request try to be clearer, because otherwise I do what you say but it doesn't work.

Reply:
Ok...trying here and there blindly I managed to do what you asked for. Hope it's okay now :)

Reply:
Note, I previously said that every time the `customFingerprint´ function is executed `.single` will be called and asked you to move it outside of that particular function.

Since the patch dependencies are being executed first, then the fingerprints are resolved, and lastly, the execute method of the patch is being called, it is necessary to previously get the resource id. This can be done by creating a dependency patch that holds the value. I will push and fix it so you can see how it needs to be done.

Reply:
> Note, I previously said that every time the `customFingerprint´ function is executed `.single` will be called and asked you to move it outside of that particular function.
> 
> Since the patch dependencies are being executed first, then the fingerprints are resolved, and lastly, the execute method of the patch is being called, it is necessary to previously get the resource id. This can be done by creating a dependency patch that holds the value. I will push and fix it so you can see how it needs to be done.

Thanks...I'll be waiting for your changes so. Sometimes how the patcher works it's dark to me. 👍